includes/class-sensei-modules.php

```diff
@@ -805,8 +808,110 @@ public function module_archive_description() {
 				echo '<p class="status ' . esc_attr( $class ) . '">' . esc_html( $status ) . '</p>';
 			}
 
-			echo '<p class="archive-description module-description">' . wp_kses_post( apply_filters( 'sensei_module_archive_description', nl2br( $module->description ), $module->term_id ) ) . '</p>';
+			if ( $this->can_view_module_content( $module, $course_id, $user_id ) ) {
+				echo '<p class="archive-description module-description">' . wp_kses_post( apply_filters( 'sensei_module_archive_description', nl2br( $module->description ), $module->term_id ) ) . '</p>';
+			}
+		}
+	}
+
+	/**
+	 * Check if we can view module content.
+	 *
+	 * @param WP_Term $module    Module term object. Defaults to the currently queried term.
+	 * @param int     $course_id Course post ID. May not be set if not viewing module in course context.
+	 * @param int     $user_id   User ID. Defaults to currently logged in user ID.
+	 *
+	 * @return bool
+	 */
+	public function can_view_module_content( WP_Term $module = null, $course_id = null, $user_id = null ) {
+		$can_view_module_content = false;
+
+		if ( null === $module ) {
+			$module = get_queried_object();
+		}
+
+		if ( ! $module instanceof WP_Term ) {
+			return false;
+		}
+
+		if ( null === $user_id ) {
+			$user_id = get_current_user_id();
+		}
+
+		if ( ! sensei_is_login_required() ) {
+			$can_view_module_content = true;
+		} elseif ( $course_id && Sensei()->course->can_access_course_content( $course_id, $user_id, 'module' ) ) {
```

Comment:
Could we remove the `elseif` here by using `||`?

Reply:
Done 79b063a0351d867b49d4349a50c51a711a039a4d